types/cache-interceptor.d.ts

```diff
@@ -88,6 +92,10 @@ declare namespace CacheHandler {
 
     get isFull (): boolean
 
+    getOrigins (): Promise<string[]>
+
+    getRoutesByOrigin (origin: string): Promise<{ method: string, path: string }[]>
```

Comment:
nit but would prefer if the `{ method: string, path: string }` type was defined outside of the fn declaration so it can be referenced elsewhere if needed (incl user space)

i.e.
```typescript
export interface RouteByOrigin {
  method: string
  path: string
}

export interface CacheStore {
  getRoutesByOrigin(origin: string): Promise<RouteByOrigin[]>
  // ...
}